src/waveform/renderers/waveformmark.cpp

```diff
@@ -254,7 +254,7 @@ struct MarkerGeometry {
         }
 
         const float increment = std::max<float>(0.f,
-                m_labelRect.height() + 2.f - std::max(0.f, 80.f - breadth));
+                static_cast<float>(m_labelRect.height()) + 2.f - std::max(0.f, 80.f - breadth));
 
```

Comment:
The threshold for this mechanism (80.f) to kick in is to be decided.

Reply:
Maybe we could make this a local variable to avoid the "magic number"?

```cpp
const float theshold = 80.f;
const float increment = std::max<float>(0.f,
        static_cast<float>(m_labelRect.height()) + 2.f - std::max(0.f, threshold - breadth));
```

Reply:
Sure. I will also change the first 0.f for a local variable minIncrement, in case we decide we want to always have a bit visible, as Joerg requests.